app/models/event.rb

```diff
@@ -66,6 +66,17 @@ def search_activities(author = nil, query_string = "", filter = "all")
   end
   alias_method :activities, :search_activities
 
+  def activities_per_day(author = nil, query_string = "", filter = "all")
+    activities = search_activities(author, query_string, filter)
+    activities.sort_by! { |activity| activity.dates.first }
+
+    activities.each_with_object(Hash.new { [] }) do |activity, grouped_by_day|
```

Comment:
Just out of curiosity: Why are you doing a `Hash.new { [] }` here? As you are using parentheses you can just write `{ [] }`, only if you don't use the parantheses Ruby doesn't know if you want to provide a hash or a block :smile: But that's nitpicky, no blocker at all :wink: